core/trino-main/src/main/java/io/trino/operator/window/pattern/MeasureComputation.java

```diff
@@ -175,6 +184,18 @@ public static Block compute(
         return work.getResult();
     }
 
+    public static Block[] precomputeNulls(List<PhysicalValueAccessor> expectedLayout)
```

Comment:
>This method doesn't really need to know about PhysicalValueAccessors. It just cares about the types

It cares about the types, but it also has to skip aggregations, which can be present in the list. 
It all happens at construction time, and `PhysicalValueAccessor`s are one of the constructor parameters, so I think we can pass just the accessors list instead of extracting the types in the constructor and passing them to the helper method.

> this doesn't seem like the right place to hold it

Why? What is the right place?